tests/functional/PendingReason.class.php

```diff
@@ -54,39 +54,93 @@ protected function testGetNextFollowupDateProvider()
-                    'bump_count'                  => 0,
-                    'last_bump_date'              => '2021-02-25 12:00:00',
+                    'pendingreason' => [
+                        'calendars_id' => 1,
+                    ],
+                    'pendingreason_item' => [
+                        'followup_frequency'          => 60,
+                        'followups_before_resolution' => 2,
+                        'bump_count'                  => 0,
+                        'last_bump_date'              => '2021-02-25 12:00:00',
+                    ]
                 ],
                 'expected' => date('2021-02-25 12:01:00')
             ],
             [
             // Case 4: second or more bump
                 'fields' => [
-                    'followup_frequency'          => 60,
-                    'followups_before_resolution' => 7,
-                    'bump_count'                  => 5,
-                    'last_bump_date'              => '2021-02-25 13:00:00',
+                    'pendingreason' => [
+                        'calendars_id' => 1,
+                    ],
+                    'pendingreason_item' => [
+                        'followup_frequency'          => 60,
+                        'followups_before_resolution' => 7,
+                        'bump_count'                  => 5,
+                        'last_bump_date'              => '2021-02-25 13:00:00',
+                    ]
                 ],
                 'expected' => '2021-02-25 13:01:00'
             ],
+            [
+            // Case 6: first with weekend
+                'fields' => [
+                    'pendingreason' => [
+                        'calendars_id' => 1,
+                    ],
+                    'pendingreason_item' => [
+                        'followup_frequency'          => 86400,
+                        'followups_before_resolution' => 7,
+                        'bump_count'                  => 0,
+                        'last_bump_date'              => '2023-04-07 13:00:00',
+                    ]
+                ],
+                'expected' => '2023-04-10 13:00:00'
+            ],
+            [
+            // Case 6: first with xmas holidays
+                'fields' => [
+                    'calendar_holiday' => [
+                        'calendars_id' => 1,
+                        'holidays_id'  => 1,
+                    ],
+                    'pendingreason' => [
+                        'calendars_id' => 1,
+                    ],
+                    'pendingreason_item' => [
+                        'followup_frequency'          => 86400,
+                        'followups_before_resolution' => 7,
+                        'bump_count'                  => 0,
+                        'last_bump_date'              => '2018-12-28 13:00:00',
+                    ]
+                ],
+                'expected' => '2019-01-07 13:00:00',
+            ],
```

Comment:
A test case to handle the special -1 value of  `followups_before_resolution` would be nice (same for `testGetAutoResolvedateProvider`).